### ISSUE TITLE
Bug: TypeError: Failed to load plugin 'jsx-a11y' declared in '.eslintrc': Invalid attempt to spread non-iterable instance.

### ISSUE THREAD

**Issue by m-nathani** [eslint/eslint#16147]
### Environment

Node version: 14.18.0
npm version:  7.24.2
Local ESLint version: 8.20.0
Global ESLint version:  -
Operating System:  linux- ubuntu 20.04

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

My eslintrc.js file

```js
module.exports = {
  // https://github.com/prettier/eslint-plugin-prettier
  extends: [
    'react-app',
    'react-app/jest',
    'airbnb',
    'plugin:prettier/recommended',
    'plugin:jsx-a11y/recommended',
  ],
  plugins: ['jsx-a11y'],
  // NOTE: need to disable the babelOptions for "babel-preset-react-app", as it was breaking due to a bug in CRA5
  // For more info: https://github.com/facebook/create-react-app/issues/12070
  // Remove the below parserOptions when its fixed
  parserOptions: {
    babelOptions: {
      presets: [['babel-preset-react-app', false], 'babel-preset-react-app/prod'],
    },
  },
  rules: {
    // NOTE: to be same as the "@umai/prettier-config" to make sure we follow the prettier rules using eslint
    'prettier/prettier': [
      'error',
      {},
      {
        usePrettierrc: true,
      },
    ],
    'import/no-unresolved': 'off',
    'import/no-named-as-default': 'off',
    'no-console': ['warn', { allow: ['warn', 'error'] }],
    'no-param-reassign': 'off',
    'function-paren-newline': 'off',
    'arrow-parens': [0, 'as-needed'],
    'arrow-body-style': ['error', 'as-needed'],
    'comma-dangle': ['error', 'only-multiline'],
    'import/no-extraneous-dependencies': 0,
    'import/prefer-default-export': 0,
    'template-curly-spacing': 'off',
    'max-len': 0,
    'react/forbid-prop-types': 0,
    'react/jsx-curly-brace-presence': 'off',
    'react/jsx-first-prop-new-line': [2, 'multiline'],
    'react/jsx-filename-extension': [1, { extensions: ['.js', '.jsx'] }],
    'react/require-default-props': 0,
    'react/self-closing-comp': 0,
    'class-methods-use-this': 0,
    'react/jsx-one-expression-per-line': 0,
    'react/function-component-definition': 0,
    'no-plusplus': 'off',
  },
  ignorePatterns: ['config/', 'scripts/', 'public/'],
};
```
</details>

Following is the my eslint npm package trace:

```cli
murtaza@murtaza-desktop:~/umai/webrms$ npm ls eslint
webrms@130.0.0 /home/murtaza/umai/webrms
├─┬ @umai/craco-plugin-micro-frontend@52.0.0
│ └─┬ react-scripts@5.0.1
│   ├─┬ eslint-webpack-plugin@3.2.0
│   │ └── eslint@8.20.0 deduped
│   ├── eslint@8.20.0 deduped
│   └─┬ react-dev-utils@12.0.1
│     └─┬ fork-ts-checker-webpack-plugin@6.5.2
│       └── eslint@8.20.0 deduped
└─┬ @umai/eslint-config@27.0.0
  ├─┬ eslint-config-airbnb@19.0.4
  │ ├─┬ eslint-config-airbnb-base@15.0.0
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ eslint-plugin-import@2.26.0
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ eslint-plugin-react-hooks@4.6.0
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ eslint-plugin-react@7.30.1
  │ │ └── eslint@8.20.0 deduped
  │ └── eslint@8.20.0 deduped
  ├─┬ eslint-config-prettier@8.5.0
  │ └── eslint@8.20.0 deduped
  ├─┬ eslint-config-react-app@7.0.1
  │ ├─┬ @babel/eslint-parser@7.18.9
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ @typescript-eslint/eslint-plugin@5.30.7
  │ │ ├─┬ @typescript-eslint/type-utils@5.30.7
  │ │ │ └── eslint@8.20.0 deduped
  │ │ ├─┬ @typescript-eslint/utils@5.30.7
  │ │ │ └── eslint@8.20.0 deduped
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ @typescript-eslint/parser@5.30.7
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ eslint-plugin-flowtype@8.0.3
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ eslint-plugin-jest@25.7.0
  │ │ ├─┬ @typescript-eslint/experimental-utils@5.30.7
  │ │ │ └── eslint@8.20.0 deduped
  │ │ └── eslint@8.20.0 deduped
  │ ├─┬ eslint-plugin-testing-library@5.5.1
  │ │ └── eslint@8.20.0 deduped
  │ └── eslint@8.20.0 deduped
  ├─┬ eslint-plugin-jsx-a11y@6.6.0
  │ └── eslint@8.20.0 deduped
  ├─┬ eslint-plugin-prettier@4.2.1
  │ └── eslint@8.20.0 deduped
  └─┬ eslint@8.20.0
    └─┬ eslint-utils@3.0.0
      └── eslint@8.20.0 deduped
```

### What did you expect to happen?

it was working in `v8.17`, now caused this error in `v8.20`. Expecting know the reason why its breaking...

Following is the custom eslint plugin i am using in my code:

```json
  "peerDependencies": {
    "eslint": "^8.3.0",
    "eslint-config-react-app": "^7.0.1"
  },
  "dependencies": {
    "@umai/prettier-config": "^11.0.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-prettier": "^8.5.0",
    "eslint-config-react-app": "^7.0.1",
    "eslint-plugin-jsx-a11y": "^6.6.0",
    "eslint-plugin-prettier": "^4.0.0"
  },

``` 

### What actually happened?

```bash
Oops! Something went wrong! :(

ESLint: 8.20.0

TypeError: Failed to load plugin 'jsx-a11y' declared in '.eslintrc » @umai/eslint-config': Invalid attempt to spread non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
Referenced from: /home/murtaza/umai/webrms/node_modules/@umai/eslint-config/index.js
    at _nonIterableSpread (/home/murtaza/umai/webrms/node_modules/@babel/runtime/helpers/nonIterableSpread.js:2:9)
    at _toConsumableArray (/home/murtaza/umai/webrms/node_modules/@babel/runtime/helpers/toConsumableArray.js:10:95)
    at Object.<anonymous> (/home/murtaza/umai/webrms/node_modules/eslint-plugin-jsx-a11y/lib/util/isInteractiveElement.js:24:61)
    at Module._compile (/home/murtaza/umai/webrms/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (/home/murtaza/umai/webrms/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/home/murtaza/umai/webrms/node_modules/eslint-plugin-jsx-a11y/lib/rules/aria-activedescendant-has-tabindex.js:22:52)
husky - pre-commit hook exited with code 1 (error)
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by dougludlow** [comment]
I just ran into this, @m-nathani did you discover the cause?

**Comment by lbottoni** [comment]
```
SyntaxError: Failed to load plugin 'jsx-a11y' declared in '--config': Unexpected token =

C:\xampp\htdocs\smyapp\node_modules\language-tags\lib\Tag.js:17
	static ERR_DEPRECATED = 1;
	
SyntaxError: Unexpected token =
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\xampp\htdocs\spindox\myapp\node_modules\language-tags\lib\index.js:11:11)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)	
```


same problem, all worked fine before install node 14.8 (i before  worked in node 10.19 for need). after install 14 and return in 10 with nvm nothing change, i always this problem.
the strange thing is, after come back in 10 i had to reinstall all node_modules because for some strange motive the build lost some ref to doblue use of componet in react
